src/librustc_expand/expand.rs

```diff
@@ -1090,6 +1092,15 @@ impl<'a, 'b> InvocationCollector<'a, 'b> {
                     .note("this may become a hard error in a future release")
                     .emit();
             }
+
+            if attr.doc_str().is_some() {
+                self.cx.parse_sess.buffer_lint_with_diagnostic(
+                    &UNUSED_DOC_COMMENTS,
+                    attr.span,
+                    ast::CRATE_NODE_ID,
```

Comment:
Is there any way to get a more specific node_id here or has this info been lost by this point? @eddyb raised the point that this will only honor crate wide attribute settings. I don't mind doing extra work to get a more specific ID here but if it can't be helped then that's fine.

Reply:
It's ok to use `CRATE_NODE_ID` here.
It's probably possible to make more specific node ids available here, but the necessary infra will be large than this whole PR.